File: ckgame-java/src/main/java/com/ckgame/politics/SchemeOutcome.java
package com.ckgame.politics;

/**
 * 阴谋月度结算结果。
 */
public final class SchemeOutcome {
    public String kind; // success / exposed / progressed
    public int schemeId;
    public SchemeKind schemeKind = null;
    public int owner = 0;
    public int target = 0;
    public double progress = 0.0;

    public SchemeOutcome(String kind, int schemeId) {
        this.kind = kind;
        this.schemeId = schemeId;
    }
}

File: ckgame-ts/src/politics/SchemeOutcome.ts
import { SchemeKind } from './SchemeKind.js';

/**
 * Scheme monthly tick outcome.
 */
export class SchemeOutcome {
  public kind: string; // 'success' | 'exposed' | 'progressed'
  public schemeId: number;
  public schemeKind: SchemeKind | null = null;
  public owner: number = 0;
  public target: number = 0;
  public progress: number = 0;

  constructor(kind: string, schemeId: number) {
    this.kind = kind;
    this.schemeId = schemeId;
  }
}
